File: benchmarks/solutions/cpp/r5-http-server.cpp
#include <httplib.h>
#include <nlohmann/json.hpp>
#include <string>

using json = nlohmann::json;

int main() {
    httplib::Server svr;

    svr.Get("/hello", [](const httplib::Request& req, httplib::Response& res) {
        std::string name = req.has_param("name") ? req.get_param_value("name") : "World";
        json body = {{"greeting", "Hello, " + name + "!"}};
        res.set_content(body.dump(), "application/json");
    });

    svr.set_default_headers({{"Content-Type", "application/json"}});
    svr.set_error_handler([](const httplib::Request&, httplib::Response& res) {
        json body = {{"error", "not found"}};
        res.status = 404;
        res.set_content(body.dump(), "application/json");
    });

    svr.listen("0.0.0.0", 8080);
    return 0;
}

File: benchmarks/solutions/typescript/r5-http-server.ts
import { createServer } from "http";

createServer((req, res) => {
  const url = new URL(req.url!, "http://localhost");
  res.setHeader("Content-Type", "application/json");

  if (url.pathname === "/hello") {
    const name = url.searchParams.get("name") || "World";
    res.writeHead(200);
    res.end(JSON.stringify({ greeting: `Hello, ${name}!` }));
  } else {
    res.writeHead(404);
    res.end(JSON.stringify({ error: "not found" }));
  }
}).listen(8080);
